### ISSUE TITLE
Allows setDate(DateTime) and similar

### ISSUE THREAD

**Issue by rentalhost** [briannesbitt/Carbon#918]
Currently `setDate()` could receives the arguments: `year`, `month?` and `day?`. Should be great if it could receives a instance of `DateTime`.

```php
// Keeps time and replace date.
$datetime = $mydate->setDate(Carbon::today());
```

**It could be applied to following methods:**

* `setDate(DateTime)`: keeps time and replace date;
* `setTime(DateTime)`: keeps date and replace time;

The `setDateTime(DateTime)` should be avoided, because it seems no make sense - maybe if it should replace date and time and keep the timezone, but I still don't guess that it makes sense.

**Comment by willrowe** [comment]
I was just looking for a simple way to copy just the date or just the time from another `DateTime`.
I like this as a possible solution, I would be willing to put together a PR for this if there is any interest.

**Comment by kylekatarnls** [comment]
Yes please @willrowe your PR would be welcome :)

**Comment by willrowe** [comment]
Great, I've been following the new development and was actually planning on submitting a PR for this anyways.

**Comment by kylekatarnls** [comment]
Fixed via #1214
